### DISCUSSION TITLE
How ListFinancialEventGroups get MarketplaceId

### DISCUSSION THREAD

**Discussion by naocan12345** [amzn/selling-partner-api-models#3125] [General]
It appears that there is one FinancialEventGroup per marketplace. However there is no direct way to identify which FinancialEventGroup belongs to which Marketplace. This can be done by inference using the OriginalTotal.CurrencyCode for some marketplaces (ie: USD is Amazon.com, CAD is Amazon.ca, GBP is Amazon.co.uk), but it doesn’t work in European marketplaces where several countries use EUR.

In order to find out the MarketplaceId for those countries, I have to make a subsequent API call to ListFinancialEvents with the FinancialEventGroupId set, then iterate through FinancialEvents to find one that has a MarketplaceName, and then use a table lookup to go back from the MarketplaceName to MarketplaceId.

However, sometimes ListFinancialEvents cannot get the MarketplaceName. What other methods can be used to get the MarketplaceId?

**Comment by jamesp26** [comment]
I've never found a way of doing it effectively. It would be lovely if FinancialEventGroup had the marketplaceid in the response.

To do this we joined orders found in a financial event group with the sales data and got the marketplace from that. Works most of the time but if you have a seller who pauses but still incurs costs it isnt bullet proof.  Would be nice to know if there is a better way.

**Reply by naocan12345** [reply]
This problem has troubled me for a long time. I haven't found a better way to get MarketplaceId, so I started this discussion. I checked a lot of information, but what I got was to judge by MarketplaceName in ListFinancialEvents, or by OriginalTotal.CurrencyCode To match, so I am troubled
